extensions/amp-nexxtv-player/0.1/amp-nexxtv-player.js

```diff
@@ -88,11 +102,26 @@ class AmpNexxtvPlayer extends AMP.BaseElement {
     Services.videoManagerForDoc(this.element).register(this);
   }
 
+  /**
+   * @param {!JsonObject} data
+   * @private 
+   */
+  playerReady_(data){
+
+    this.playerReadyResolver_(this.iframe_);
+
+    dev().info(
+      TAG,
+      'nexx player ready',
+      data
+    );
+  }
+
   /**
    * @return {string}
    * @private
    */
-  resolveVideoIframeSrc_() {
+  getVideoIframeSrc_() {
```

Comment:
Consider `Element.dataset` and `addParamsToUrl`. This method could clean up nicely:

```js
  getVideoIframeSrc_() {
    const {
      client,
      domainId,
      mediaid,
      mode,
      streamtype,
      disableAds,
      streamingFilter,
      exitMode,
    } = el.dataset;

    const mediaId = userAssert(
      mediaid,
      'The data-mediaid attribute is required for <amp-nexxtv-player> %s'
    );

    const clientId = userAssert(
      client || domainId,
      'One of data-client or data-domain-id attributes is required for <amp-nexxtv-player> %s',
      el
    );

    const url =
      this.origin_ +
      [clientId, streamtype, mediaid].map(encodeURIComponent).join('/');

    return assertAbsoluteHttpOrHttpsUrl(
      addParamsToUrl(
        url,
        dict({
          platform: 'amp',
          disableAds: disableAds,
          streamingFilter: streamingFilter,
          exitMode: exitMode,
          consentString: this.consentString_,
        })
      )
    );
  }
```

Reply:
I've reworked this part.